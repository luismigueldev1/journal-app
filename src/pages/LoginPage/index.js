import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import {
  loginWithEmailAndPasswordAction,
  loginWithGoogleProvider,
} from "../../actions/authActions";
import { removeErrorAction } from "../../actions/uiActions";
import { useForm } from "../../hooks/useForm";

export default function LoginPage() {
  const [values, handleInputChange] = useForm({
    email: "luismiguel@gmail.com",
    password: "123456",
  });

  const dispatch = useDispatch();
  const { loading } = useSelector((state) => state.ui);
  const { error } = useSelector((state) => state.ui);

  const handleLogin = (event) => {
    event.preventDefault();
    dispatch(loginWithEmailAndPasswordAction(values.email, values.password));
  };

  const handleGoogleLogin = (event) => {
    event.preventDefault();
    dispatch(loginWithGoogleProvider());
  };

  useEffect(() => {
    return () => {
      dispatch(removeErrorAction());
    };
  }, [dispatch]);
  return (
    <div className="auth__container">
      <h3 className="text--center">Login Page</h3>

      {error && (
        <div className="alert alert-danger">
          <p className="alert-title ">{error}</p>
        </div>
      )}

      <form
        onSubmit={handleLogin}
        className="animate__animated animate__fadeIn animate__faster"
      >
        <div className="form__control">
          <label htmlFor="email">Email</label>
          <input
            type="text"
            name="email"
            value={values.email}
            onChange={handleInputChange}
          />
        </div>
        <div className="form__control">
          <label htmlFor="password">Password</label>
          <input
            type="password"
            name="password"
            value={values.password}
            onChange={handleInputChange}
          />
        </div>
        <div className="form__control">
          <button type="submit" className="pointer" disabled={loading}>
            {loading ? "Loading.." : "Login"}
          </button>
        </div>

        <hr />
        <div className="google__login pointer">
          <p>
            <b>Login with social:</b>
          </p>
          <button
            className="pointer mt-1 google-login-button"
            onClick={handleGoogleLogin}
          >
            <picture>
              <img src="/assets/google-logo.png" alt="Google Login" />
            </picture>

            <p>
              <b>Sign in with google</b>
            </p>
          </button>
        </div>
        <div className="create__account">
          <Link to="/auth/register">Create an account</Link>
        </div>
      </form>
    </div>
  );
}
