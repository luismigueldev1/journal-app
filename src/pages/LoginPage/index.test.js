import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import { mount } from "enzyme";
import LoginPage from ".";
import {
  loginWithGoogleProvider,
  loginWithEmailAndPasswordAction,
} from "../../actions/authActions";
jest.mock("../../actions/authActions", () => {
  return {
    loginWithGoogleProvider: jest.fn(),
    loginWithEmailAndPasswordAction: jest.fn(),
  };
});

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  auth: {},
  ui: { loading: false, error: null },
  notes: [],
};

let store = mockStore(initialState);
store.dispatch = jest.fn();

describe("Pruebas en <LoginScreen/>", () => {
  beforeEach(() => {
    store = mockStore(initialState);
    jest.clearAllMocks();
  });

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <LoginPage />
      </MemoryRouter>
    </Provider>
  );
  test("debe de mostrarse correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de disparar la accion de loginWithGoogleProvider ", () => {
    wrapper.find(".google-login-button").simulate("click");
    expect(loginWithGoogleProvider).toHaveBeenCalled();
  });

  test("debe de disparar la accion de loginWithEmailAndPasswordAction ", () => {
    wrapper.find("form").simulate("submit");
    expect(loginWithEmailAndPasswordAction).toHaveBeenLastCalledWith(
      "luismiguel@gmail.com",
      "123456"
    );
    expect(loginWithEmailAndPasswordAction).toHaveBeenCalledTimes(1);
  });
});
