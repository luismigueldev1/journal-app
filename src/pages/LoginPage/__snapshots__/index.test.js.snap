// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> debe de mostrarse correctamente  1`] = `
<div
  className="auth__container"
>
  <h3
    className="text--center"
  >
    Login Page
  </h3>
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <div
      className="form__control"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value="luismiguel@gmail.com"
      />
    </div>
    <div
      className="form__control"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value="123456"
      />
    </div>
    <div
      className="form__control"
    >
      <button
        className="pointer"
        disabled={false}
        type="submit"
      >
        Login
      </button>
    </div>
    <hr />
    <div
      className="google__login pointer"
    >
      <p>
        <b>
          Login with social:
        </b>
      </p>
      <button
        className="pointer mt-1 google-login-button"
        onClick={[Function]}
      >
        <picture>
          <img
            alt="Google Login"
            src="/assets/google-logo.png"
          />
        </picture>
        <p>
          <b>
            Sign in with google
          </b>
        </p>
      </button>
    </div>
    <div
      className="create__account"
    >
      <a
        href="/auth/register"
        onClick={[Function]}
      >
        Create an account
      </a>
    </div>
  </form>
</div>
`;
